sympy/physics/control/tests/test_lti.py

```diff
@@ -0,0 +1,233 @@
+        Matrix([5 / (s + 2)])
+    )
+    H = TransferFunctionMatrix(
+        Matrix([7 / (p**9 + 9*p) + 5 / (s + 2)])
+    )
+    assert together(H1.parallel(H2).G) == together(H.G)
+
+
+def test_TransferFunctionMatrix_negate():
+    G1 = Matrix([s / (s + 1)])
+    tf1 = TransferFunctionMatrix(G1)
+    assert tf1.neg() == TransferFunctionMatrix(
+        Matrix([-s / (s + 1)])
+    )
+
+    G2 = Matrix([[(4*p - 10) / (2*p + 1), 3 / (p + 2)],
+            [1 / (2*p**2 + 5*p + 2), (p + 1) / (p**2 + 4*p + 4)]])
+    tf2 = TransferFunctionMatrix(G2)
+    assert tf2.neg() == TransferFunctionMatrix(
+        Matrix([[(10 - 4*p) / (2*p + 1),  -3 / (p + 2)],
+            [-1 / (2*p**2 + 5*p + 2), (-p - 1) / (p**2 + 4*p + 4)]])
+    )
+
+
+def test_TransferFunctionMatrix_is_proper():
+    G1 = Matrix([2*p / (2 + p**2 - p), p / (1 + p**3)])
+    assert TransferFunctionMatrix(G1).is_proper
+
+    G2 = Matrix([2*s**6/(2 + s**2 - s), (s**4)/(1 + s**2)])
+    assert not TransferFunctionMatrix(G2).is_proper
+
+
+def test_TransferFunctionMatrix_is_strictly_proper():
+    G1 = Matrix([2*s**6 / (2 + s**2 - s), (s**4) / (1 + s**2)])
+    assert not TransferFunctionMatrix(G1).is_strictly_proper
+
+    G2 = Matrix([[(4*p - 10) / (2*p**3 + 1), 3 / (p + 2)],
+            [1 / (2*p**2 + 5*p + 2), (p + 1) / (p**2 + 4*p + 4)]])
+    assert TransferFunctionMatrix(G2).is_strictly_proper
+
+
+def test_TransferFunction_construction():
+    tf = TransferFunction(s + 1, s**2 + s + 1)
+    assert tf.num == (s + 1)
+    assert tf.den == (s**2 + s + 1)
+    assert tf.args == (s + 1, s**2 + s + 1)
+
+    tf1 = TransferFunction(s + 4, s - 5)
+    assert tf1.num == (s + 4)
+    assert tf1.den == (s - 5)
+    assert tf1.args == (s + 4, s - 5)
+
+    # using different polynomial variables.
+    tf2 = TransferFunction(p + 3, p**2 - 9)
+    assert tf2.num == (p + 3)
+    assert tf2.den == (p**2 - 9)
+    assert tf2.args == (p + 3, p**2 - 9)
+
+    tf3 = TransferFunction(p**3 + 5*p**2 + 4, p**4 + 3*p + 1)
+    assert tf3.args == (p**3 + 5*p**2 + 4, p**4 + 3*p + 1)
+
+    # no pole-zero cancellation on its own.
+    tf4 = TransferFunction((s + 3)*(s - 1), (s - 1)*(s + 5))
+    assert tf4.den == (s**2 + 4*s - 5)
+    assert tf4.args == (s**2 + 2*s - 3, s**2 + 4*s - 5)
+
+    # purely symbolic polynomials.
+    tf5 = TransferFunction(p, p)
+    assert tf5.num == tf5.den == p
+    assert tf5.args == (p, p)
+
+    tf6 = TransferFunction(5, 6)
+    assert tf6.num == 5
+    assert tf6.den == 6
+    assert tf6.args == (5, 6)
+
+    # ValueError when denominator is zero.
+    raises(ValueError, lambda: TransferFunction(4, 0))
+    raises(ValueError, lambda: TransferFunction(s, 0))
+    raises(ValueError, lambda: TransferFunction(0, 0))
+
+
+def test_TransferFunction_functions():
+    # explicitly cancel poles and zeros.
+    tf0 = TransferFunction(s**5 + s**3 + s, s - s**2)
+    a = TransferFunction(-s**4 - s**2 - 1, s - 1)
+    assert tf0.cancel_poles_and_zeros() == a
```

Comment:
I think that `tf.simplify()` might be a better name because this is already a common thing for sympy objects. It should apply simplification hints that are specific for simplifying ratios of polynomials under the hood. `simplify(tf)` should also work.

Reply:
If you don't call it simplify or use simplify() what is the difference that `cancel_poles_and_zeros()` would do that simplify doesnt?

Reply:
okay, I think I wanted to have another method `cancel_poles_and_zeros()` as it would internally use `simplify` and `expand` to give us the desired result after cancellation. Like this for example: 
``` python
>>> from sympy import simplify
>>> tf = TransferFunction(s**5 + s**3 + s, s - s**2)
>>> simplify(tf) 
TransferFunction(-(s**4 + s**2 + 1), s - 1)  
>>> tf.cancel_poles_and_zeros() # distributes the -ve sign.
TransferFunction(-s**4 - s**2 - 1, s - 1)
```
Maybe I should rename it to `simplify`?